.github/workflows/test.yml

```diff
@@ -2,11 +2,25 @@ name: CI
 
 on:
   push:
+    branches:
+      - main
   pull_request:
 
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
```

Comment:
**Question:** why use Python 3.8 when there are newer versions available?

Reply:
was erring on the side of cautiousness, especially for black and its target/available Python. But then https://github.com/psf/black/releases/tag/23.1.0 makes that obsolete and you can run the whole thing with a newer Python version and it will automagically detech all target versions it needs from config. Updated to run with 3.11